include/readout/models/IterableQueueModel.hpp

```diff
@@ -323,23 +340,16 @@ struct IterableQueueModel : public LatencyBufferConcept<T>
     pointer operator->() { return &m_queue.records_[m_index]; }
     Iterator& operator++() // NOLINT(runtime/increment_decrement) :)
     {
-      if (good()) {
-        m_index++;
-        if (m_index == m_queue.size_) {
-          m_index = 0;
-        }
-      }
-      if (!good()) {
-        m_index = std::numeric_limits<uint32_t>::max(); // NOLINT(build/unsigned)
+      m_index++;
+      if (m_index == m_queue.size_) {
+        m_index = 0;
       }
       return *this;
     }
-    Iterator operator++(int amount) // NOLINT(runtime/increment_decrement) :)
+    Iterator operator++(int) // NOLINT(runtime/increment_decrement) :)
     {
       Iterator tmp = *this;
-      for (int i = 0; i < amount; ++i) {
-        ++(*this);
```

Comment:
This was a critical bugfix in the IterableQueueModel, that the PR attempts to remove. Something is not right.